Matchismo/Matchismo/Model/SetCard.mm

```diff
@@ -0,0 +1,165 @@
+//
+//  SetCard.m
+//  Matchismo
+//
+//  Created by Otto Olkkonen on 15/06/2022.
+//
+
+#include "SetCard.h"
+
+@interface SetCard()
+- (BOOL)allSameNoOfShapes:(NSArray<SetCard *> *)otherCards;
+- (BOOL)allDifferentNoOfShapes:(NSArray<SetCard *> *)otherCards;
+- (BOOL)allSameShape:(NSArray<SetCard *> *)otherCards;
+- (BOOL)allDifferentShape:(NSArray<SetCard *> *)otherCards;
+- (BOOL)allSameShading:(NSArray<SetCard *> *)otherCards;
+- (BOOL)allDifferentShading:(NSArray<SetCard *> *)otherCards;
+- (BOOL)allSameColor:(NSArray<SetCard *> *)otherCards;
+- (BOOL)allDifferentColor:(NSArray<SetCard *> *)otherCards;
+@end
+
+@implementation SetCard
+
+- (NSMutableDictionary *)attributes {
+  NSMutableDictionary *att = [@{NSForegroundColorAttributeName:self.color}
+                                     mutableCopy];
+  
+  switch (self.shading) {
+    case setCardShadingSolid:
+      att[NSStrokeWidthAttributeName] = @-5;
+      break;
+    case setCardShadingStriped:
+      att[NSStrokeWidthAttributeName] = @-5;
+      att[NSStrikethroughStyleAttributeName] = [NSNumber numberWithInt:
+                                                       NSUnderlineStyleDouble];
+      att[NSStrikethroughColorAttributeName] = UIColor.whiteColor;
+      break;
+    case setCardShadingOpen:
+      att[NSStrokeWidthAttributeName] = @5;
+      break;
+    default:
+      break;
+  }
+  
+  return att;
+}
+
+- (NSAttributedString *)contents {
+
+  auto text = [NSString stringWithFormat:@"%llu %@",
+                 (long long)self.noOfShapes,
+                 self.shape];
+  NSMutableDictionary *attributes = [@{NSForegroundColorAttributeName:self.color}
+                                     mutableCopy];
+  auto *attributedString = [[NSMutableAttributedString alloc]
+                                                initWithString:text attributes:attributes];
+  const auto range = [[attributedString string] rangeOfString:self.shape];
+  [attributedString setAttributes:self.attributes range:range];
+ 
+  return attributedString;
+}
+
+- (BOOL)allSameNoOfShapes:(NSArray<SetCard *> *)otherCards {
+  for (SetCard *card in otherCards) {
+    if (card.noOfShapes != self.noOfShapes) {
+      return NO;
+    }
+  }
+  return YES;
+}
+
+- (BOOL)allDifferentNoOfShapes:(NSArray<SetCard *> *)otherCards {
+  NSMutableSet *set = [[[NSSet alloc] init] mutableCopy];
+  [set addObject:@(self.noOfShapes)];
```

Comment:
We're just adding one object to the set here, and then never changing it.
Looks like a mistake?